java-processors/src/main/java/com/dataiku/dip/plugins/tradearea/TradeAreaProcessor.java

```diff
@@ -0,0 +1,217 @@
+                    "Select the unit of distances from:\n" +
+                    "* Kilometers\n" +
+                    "* Miles\n\n" +
+                    "Select the shape of trade area from:\n" +
+                    "* Rectangular\n" +
+                    "* Circular\n\n";
+        }
+
+        @Override
+        public Class<Parameter> stepParamClass() {
+            return Parameter.class;
+        }
+
+        @Override
+        public ProcessorDesc describe() {
+            // UI creation using ParamDesc
+            return ProcessorDesc.withGenericForm(this.getName(), actionVerb("Create") + " trade area zone")
+                    .withMNEColParam("inputColumn", "Input column")
+                    .withColParam("outputColumn", "Output column")
+                    .withParam(ParamDesc.advancedSelect("unitMode", "Distance unit", "", UnitMode.class).withDefaultValue(UnitMode.KILOMETERS))
+                    .withParam(ParamDesc.advancedSelect("shapeMode", "Shape", "", ShapeMode.class).withDefaultValue(ShapeMode.CIRCLE))
+                    .withParam(ParamDesc.doubleP("radius", "Radius"))
+                    .withParam(ParamDesc.doubleP("width", "Width"))
+                    .withParam(ParamDesc.doubleP("height", "Height"));
+        }
+
+        @Override
+        public TradeAreaProcessor build(TradeAreaProcessor.Parameter parameter) throws Exception {
+            return new TradeAreaProcessor(parameter);
+        }
+    };
+
+    public TradeAreaProcessor(Parameter params) {
+        this.params = params;
+    }
+
+    private final Parameter params;
+    private Column outputColumn;
+    private Column cd;
+    private AreaGenerator generator;
+
+    @Override
+    public void processRow(Row row) throws Exception {
+        String str = row.get(cd);
+        // Handle null in input
+        if (str == null || str.equals("")) {
+            getProcessorOutput().emitRow(row);
+            return;
+        }
+
+        String output = "";
+        // Actual row to row processing
+        // str should be a WKT description of the input geospatial point
+        // example: POINT(-73.9723 40.64749) where -73.9723 is a longitude, 40.64749 is a latitude
+        GeoPoint.Coords coords = GeoPoint.convert(str);
+        if (coords != null){
+            output = generator.generateArea(coords);
+        }
+
+        // Handle null in output
+        if (output != null) {
+            row.put(outputColumn, output);
+        }
+
+        getProcessorOutput().emitRow(row);
+    }
+
+    @Override
+    public void postProcess() throws Exception {
+        getProcessorOutput().lastRowEmitted();
+    }
+
+    @Override
+    public void init() throws Exception {
+        cd = getCf().column(params.inputColumn, ProcessorRole.INPUT_COLUMN);
+        if (!StringUtils.isBlank(params.outputColumn)) {
+            outputColumn = getCf().column(params.outputColumn, ProcessorRole.OUTPUT_COLUMN);
+        } else {
+            outputColumn = cd;
```

Comment:
There is a bug here.
Let say I have three columns in my dataset (lon, lat and location).
I choose lon as an input col, and no output column, then the value of the lon columns are wiped out.
Cf screenshot below:
<img width="1387" alt="Capture d’écran 2021-02-08 à 10 43 50 PM" src="https://user-images.githubusercontent.com/54799490/107285239-262f3f00-6a5f-11eb-8a85-98f4e7772b9f.png">

Reply:
Can't reproduce the bug.

Reply:
<img width="470" alt="Screenshot 2021-02-15 at 11 25 18" src="https://user-images.githubusercontent.com/17903135/107934540-7d15a680-6f80-11eb-9364-173e68be5a72.png">

Reply:
Fixed in 1c592f4.